### PR TITLE
Fixes #4438: wrong cast used when setting Lua mixer script input source

### LINKED ISSUE #4438: Mixer script source selector only has a few telemetry values
Mixer script source selector only has a few telemetry values. I only see 3 while i have about 20 sensors....

### PR DESCRIPTION
**Pull request by projectkk2glider**
Improved parsing of Lua mixer script inputs

Closes #4438

### CODE REVIEW

File: radio/src/gui/128x64/model_custom_scripts.cpp [hunk 1]
```diff
@@ -96,14 +96,14 @@ void menuModelCustomScriptOne(event_t event)
...
-      if (scriptInputsOutputs[s_currIdx].inputs[inputIdx].type == 0) {
+      if (scriptInputsOutputs[s_currIdx].inputs[inputIdx].type == INPUT_TYPE_VALUE) {
         lcdDrawNumber(SCRIPT_ONE_2ND_COLUMN_POS, y, g_model.scriptsData[s_currIdx].inputs[inputIdx]+scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, attr|LEFT);
         if (attr) {
           CHECK_INCDEC_MODELVAR(event, g_model.scriptsData[s_currIdx].inputs[inputIdx], scriptInputsOutputs[s_currIdx].inputs[inputIdx].min-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, scriptInputsOutputs[s_currIdx].inputs[inputIdx].max-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def);
         }
       }
       else {
-        uint8_t *source = (uint8_t *)&g_model.scriptsData[s_currIdx].inputs[inputIdx];
+        uint16_t * source = (uint16_t *)&g_model.scriptsData[s_currIdx].inputs[inputIdx];
```
Comment (3djc) [inline radio/src/gui/128x64/model_custom_scripts.cpp]: Would source_t be better suited ?

File: radio/src/gui/128x64/model_custom_scripts.cpp [hunk 2]
```diff
@@ -96,14 +96,15 @@ void menuModelCustomScriptOne(event_t event)
...
-      if (scriptInputsOutputs[s_currIdx].inputs[inputIdx].type == 0) {
+      if (scriptInputsOutputs[s_currIdx].inputs[inputIdx].type == INPUT_TYPE_VALUE) {
         lcdDrawNumber(SCRIPT_ONE_2ND_COLUMN_POS, y, g_model.scriptsData[s_currIdx].inputs[inputIdx]+scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, attr|LEFT);
         if (attr) {
           CHECK_INCDEC_MODELVAR(event, g_model.scriptsData[s_currIdx].inputs[inputIdx], scriptInputsOutputs[s_currIdx].inputs[inputIdx].min-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, scriptInputsOutputs[s_currIdx].inputs[inputIdx].max-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def);
         }
       }
       else {
-        uint8_t *source = (uint8_t *)&g_model.scriptsData[s_currIdx].inputs[inputIdx];
+        int16_t * source = &g_model.scriptsData[s_currIdx].inputs[inputIdx];
```
Comment (bsongis) [inline radio/src/gui/128x64/model_custom_scripts.cpp]: why signed? and why not source_t?

Comment (projectkk2glider) [inline radio/src/gui/128x64/model_custom_scripts.cpp]: `g_model.scriptsData[s_currIdx].inputs[inputIdx]` is `int16_t`. Not sure it is wise to cast it to the unsigned pointer.

Comment (bsongis) [inline radio/src/gui/128x64/model_custom_scripts.cpp]: Yes I think so, this inputs field should even be an union in my opinion

File: radio/src/gui/128x64/model_custom_scripts.cpp [hunk 3]
```diff
@@ -97,17 +97,18 @@ void menuModelCustomScriptOne(event_t event)
...
-          CHECK_INCDEC_MODELVAR(event, g_model.scriptsData[s_currIdx].inputs[inputIdx], scriptInputsOutputs[s_currIdx].inputs[inputIdx].min-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, scriptInputsOutputs[s_currIdx].inputs[inputIdx].max-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def);
+          CHECK_INCDEC_MODELVAR(event, g_model.scriptsData[s_currIdx].inputs[inputIdx].value, \
+                                scriptInputsOutputs[s_currIdx].inputs[inputIdx].min-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, \
+                                scriptInputsOutputs[s_currIdx].inputs[inputIdx].max-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def);
         }
       }
       else {
-        int16_t * source = &g_model.scriptsData[s_currIdx].inputs[inputIdx];
-        static_assert(sizeof(*source) == sizeof(g_model.scriptsData[0].inputs[0]), "sizes must be same");
-        drawSource(SCRIPT_ONE_2ND_COLUMN_POS, y, *source + scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, attr);
```
Comment (projectkk2glider) [inline radio/src/gui/128x64/model_custom_scripts.cpp]: We had a bug here `def` is not to be used with source type.

File: radio/src/gui/128x64/model_custom_scripts.cpp [hunk 4]
```diff
@@ -97,17 +97,18 @@ void menuModelCustomScriptOne(event_t event)
...
+        drawSource(SCRIPT_ONE_2ND_COLUMN_POS, y, g_model.scriptsData[s_currIdx].inputs[inputIdx].source , attr);
         if (attr) {
-          CHECK_INCDEC_MODELSOURCE(event, *source, scriptInputsOutputs[s_currIdx].inputs[inputIdx].min-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def, scriptInputsOutputs[s_currIdx].inputs[inputIdx].max-scriptInputsOutputs[s_currIdx].inputs[inputIdx].def);
```
Comment (projectkk2glider) [inline radio/src/gui/128x64/model_custom_scripts.cpp]: Another bug: min and max make no sense here.

**Comment by bsongis** [comment]
All good, thanks!

**Pull request merged**